ac/ac-ranking/src/ActivityRunner/Answer.jsx

```diff
@@ -0,0 +1,93 @@
+// @flow
+
+import * as React from 'react';
+import { A } from 'frog-utils';
+import { findKey } from 'lodash';
+import { Badge, Glyphicon, ListGroupItem } from 'react-bootstrap';
+
+import { getXYFromRanking } from '../Dashboard';
+
+const blue = '#337ab7';
+const grey = '#d3d3d3';
+
+const nKey = x => Object.keys(x).length;
+
+const changeRank = (props, incr) => {
+  const {
+    activityData: { config },
+    logger,
+    dataFn,
+    userInfo,
+    data,
+    answer
+  } = props;
+  const { answers } = data;
+
+  const ans = answers[userInfo.id];
+
+  if (
+    !(ans[answer] === 1 && incr < 0) &&
+    !(ans[answer] === nKey(ans) && incr > 0)
+  ) {
+    const newAnswers = { ...ans };
+    const switchID = findKey(ans, x => x === ans[answer] + incr);
+    newAnswers[answer] += incr;
+    newAnswers[switchID] -= incr;
+
+    logger([
+      {
+        type: 'listOrder',
+        itemId: answer,
+        value: JSON.stringify(newAnswers)
+      },
+      {
+        type: 'coordinates',
+        payload: getXYFromRanking(newAnswers, config)
+      }
+    ]);
+    dataFn.objInsert(newAnswers, ['answers', userInfo.id]);
+  }
+};
+
+export default (props: Object) => {
+  const { answer, memAnswers, userID, uiID } = props;
+  return (
+    <ListGroupItem>
+      <font size={4}>
+        {userID.toString() === uiID.toString() && (
```

Comment:
do you need this? userId should be a string always. With Louis' change, it will be a string in preview too

Reply:
UserID was a string but uiID was a number, I had put it in because it seemed like it could not be guaranteed.